From 59d3efdf9e55958c6a3651e8eda2a9d6fe48e192 Mon Sep 17 00:00:00 2001
From: Agriya Khetarpal <74401230+agriyakhetarpal@users.noreply.github.com>
Date: Fri, 9 Aug 2024 19:00:41 +0530
Subject: [PATCH 14/18] Skip svd_gesdd test

This patch excludes a test for gesdd which was introduced in this PR:
https://github.com/scipy/scipy/pull/20349. It is not useful for Pyodide
since it is a memory-intensive test and it is not expected to pass in
a WASM environment where allocating memory for large arrays is tricky.

This patch has been upstreamed in https://github.com/scipy/scipy/pull/21349
and it can be safely removed once SciPy v1.15.0 is released and is being
integrated in Pyodide.

---
 scipy/linalg/tests/test_decomp.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/scipy/linalg/tests/test_decomp.py b/scipy/linalg/tests/test_decomp.py
index b43016c027..cbd80252b1 100644
--- a/scipy/linalg/tests/test_decomp.py
+++ b/scipy/linalg/tests/test_decomp.py
@@ -1,5 +1,6 @@
 import itertools
 import platform
+import sys
 
 import numpy as np
 from numpy.testing import (assert_equal, assert_almost_equal,
@@ -37,6 +38,8 @@ try:
 except ImportError:
     CONFIG = None
 
+IS_WASM = (sys.platform == "emscripten" or platform.machine() in ["wasm32", "wasm64"])
+
 
 def _random_hermitian_matrix(n, posdef=False, dtype=float):
     "Generate random sym/hermitian array of the given size n"
@@ -1179,6 +1182,9 @@ class TestSVD_GESVD(TestSVD_GESDD):
     lapack_driver = 'gesvd'
 
 
+# Allocating an array of such a size leads to _ArrayMemoryError(s)
+# since the maximum memory that can be in 32-bit (WASM) is 4GB
+@pytest.mark.skipif(IS_WASM, reason="out of memory in WASM")
 @pytest.mark.fail_slow(5)
 def test_svd_gesdd_nofegfault():
     # svd(a) with {U,VT}.size > INT_MAX does not segfault
-- 
2.39.3 (Apple Git-146)

